### DISCUSSION TITLE
Importing comments from Drupal 6

### DISCUSSION THREAD

**Discussion by mipimipi** [umputun/remark42#905] [Q&A] [answered]
Hi,
I am currently converting a blog from Drupal 6 to Hugo. The blog contains several thousands of comments. What is the best way to import them to remark42? My first approach would be to generate a XML file in the Disqus export format and to import that file into remark42. It there a better way?

The users that created the comments are of 3 categories: 
1. Users with an account for my blog
1. Users without account but with a known mail address
1. None of them (i.e. really anonymous users)

In remark42 I do not want to allow comments from anonymous users but want to allow registrations via mail. Is there a way to create user accounts in remark42 for the first two categories in an automated fashion?

In the Drupal comments, people could add the url of their own blogs / web sites. Is there a way to import them to remark42 (would be a pity if this information would get lost)?

BTW: the rest of my blog (i.e. all the posts) is already converted to Hugo. Only the comments are missing.

Thanks, mipi

**Comment by umputun** [comment, marked as answer]
If you export comments to either Disqus-like XML or native json, you could import them directly to remark42. There are no categories of any kind in remark42 and, in fact, you don't create accounts at all. Email access doesn't involve any account creation.

As I moved my blogs from Disqus back in days, I got all the comments with proper user names, however, remark42 doesn't support any additional meta-info about users, except avatars. So, there is no place to keep URLs in the user info.

**Comment by mipimipi** [comment]
Thanks. One more question: How can I set the id that remark42 uses to identify the comment belonging to a certain Hugo page? Since I imported all the posts and comments from a Drupal site, I have to take care of this explicitly.

**Reply by paskal** [reply]
Please take a look here: https://remark42.com/docs/latest/migration/

I think that you'll need to import them with Drupal URLs and then remap to new Hugo URLs (if they are changed) used the doc I provided.

**Reply by mipimipi** [reply]
Thanks. I deduce from what you wrote that the remark42 import logic for disqus comments takes the thread id of a comment, determines the corresponding thread url and takes this to link the comment with Hugo blog post. Correct? In this case I just have to set the correct thread url during the generation of the Disqus XML file that I plan to import to remark42. I will try ...

**Reply by mipimipi** [reply]
It worked :-) I imported all comments. Thanks for the support.
